mm/2s2h/Rando/Logic/Regions.cpp

```diff
@@ -832,6 +835,89 @@ std::unordered_map<RandoRegionId, RandoRegion> Regions = {
+        .checks = {
+            CHECK(RC_MOON_FIERCE_DEITY_MASK, true),
+        },
+        .exits = { //     TO                                         FROM
+            EXIT(ENTRANCE(CLOCK_TOWER_ROOFTOP, 0),                   ONE_WAY_EXIT, true),
+            EXIT(ENTRANCE(MOON_DEKU_TRIAL, 0),              ENTRANCE(THE_MOON, 0), true),
+            EXIT(ENTRANCE(MOON_GORON_TRIAL, 0),             ENTRANCE(THE_MOON, 0), true),
+            EXIT(ENTRANCE(MOON_LINK_TRIAL, 0),              ENTRANCE(THE_MOON, 0), true),
+            EXIT(ENTRANCE(MOON_ZORA_TRIAL, 0),              ENTRANCE(THE_MOON, 0), true),
+            EXIT(ENTRANCE(MAJORAS_LAIR, 0),                          ONE_WAY_EXIT, true),
+        },
+        .oneWayEntrances = {
+            ENTRANCE(THE_MOON, 0),
+        },
+    } },
+    { RR_MOON_DEKU_TRIAL, RandoRegion{ .sceneId = SCENE_LAST_DEKU,
+        .checks = {
+            CHECK(RC_MOON_TRIAL_DEKU_HP, CAN_BE_DEKU),
+        },
+        .exits = { //     TO                                         FROM
+            EXIT(ENTRANCE(THE_MOON, 0),                     ENTRANCE(MOON_DEKU_TRIAL, 0), true),
+            EXIT(ENTRANCE(THE_MOON, 0),                     ONE_WAY_EXIT, CAN_BE_DEKU), // Exit from Mask NPC
+        },
+    } },
+    { RR_MOON_GORON_TRIAL, RandoRegion{ .sceneId = SCENE_LAST_GORON,
+        .checks = {
+            CHECK(RC_MOON_TRIAL_GORON_HP, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_EARLY_1, true),
+            CHECK(RC_MOON_TRIAL_GORON_POT_EARLY_2, true),
+            CHECK(RC_MOON_TRIAL_GORON_POT_EARLY_3, true),
+            CHECK(RC_MOON_TRIAL_GORON_POT_EARLY_4, true),
+            CHECK(RC_MOON_TRIAL_GORON_POT_1, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_10, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_11, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_2, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_3, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_4, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_5, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_6, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_7, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_8, CAN_BE_GORON && HAS_MAGIC),
+            CHECK(RC_MOON_TRIAL_GORON_POT_9, CAN_BE_GORON && HAS_MAGIC),
+        },
+        .exits = { //     TO                                         FROM
+            EXIT(ENTRANCE(THE_MOON, 0),                     ENTRANCE(MOON_GORON_TRIAL, 0), true),
+            EXIT(ENTRANCE(THE_MOON, 0),                     ONE_WAY_EXIT, CAN_BE_GORON && HAS_MAGIC), // Exit from Mask NPC
+        },
+    } },
+    { RR_MOON_LINK_TRIAL, RandoRegion{ .sceneId = SCENE_LAST_LINK,
+        .checks = {
+            CHECK(RC_MOON_TRIAL_LINK_GARO_MASTER_CHEST, HAS_ITEM(ITEM_HOOKSHOT)),
```

Comment:
Maybe we should use the CAN_KILL_GARO_MASTER requirement here since its already defined. Same with CAN_KILL_IRONKNUCKLE in ln.892.